Public/Src/Cache/ContentStore/Distributed/Sessions/DistributedContentSession.cs

```diff
@@ -283,42 +283,57 @@ private Task<ProactiveCopyResult> RequestProactiveCopyIfNeededAsync(OperationCon
                             return ProactiveCopyResult.CopyNotRequiredResult;
                         }
 
-                        IReadOnlyList<MachineLocation> buildRingMachines;
+                        IReadOnlyList<MachineLocation> buildRingMachines = null;
 
                         // Get random machine inside build ring
                         Task<BoolResult> insideRingCopyTask;
-                        if (_buildIdHash != null)
+                        if (Settings.ProactiveCopyMode.HasFlag(ProactiveCopyMode.InsideRing))

```

Comment:
Keep in mind that this operation causes 2 boxing allocations.

so an alternative (if you want) is if (Settings.ProactiveCopyMode & ProactiveCopyMode.InsideRing != 0).